demo.ts

```diff
@@ -1,85 +1,96 @@
+import { loadSync, Status } from "./dev_deps.ts";
 import {
-  createAuth0OAuth2Client,
-  createDiscordOAuth2Client,
-  createDropboxOAuth2Client,
-  createFacebookOAuth2Client,
-  createGitHubOAuth2Client,
-  createGitLabOAuth2Client,
-  createGoogleOAuth2Client,
-  createNotionOAuth2Client,
-  createOktaOAuth2Client,
-  createPatreonOAuth2Client,
-  createSlackOAuth2Client,
-  createSpotifyOAuth2Client,
-  createTwitterOAuth2Client,
+  createAuth0OAuthConfig,
+  createDiscordOAuthConfig,
+  createDropboxOAuthConfig,
+  createFacebookOAuthConfig,
+  createGitHubOAuthConfig,
+  createGitLabOAuthConfig,
+  createGoogleOAuthConfig,
+  createNotionOAuthConfig,
+  createOktaOAuthConfig,
+  createPatreonOAuthConfig,
+  createSlackOAuthConfig,
+  createSpotifyOAuthConfig,
+  createTwitterOAuthConfig,
   getSessionAccessToken,
   getSessionId,
   handleCallback,
+  type OAuthConfig,
+  type OAuthUserConfig,
   signIn,
   signOut,
 } from "./mod.ts";
 
 loadSync({ export: true });
 
+type CreateOAuthConfigFn = (config: OAuthUserConfig) => OAuthConfig;
+
+const providers: Record<string, CreateOAuthConfigFn> = {
+  Auth0: createAuth0OAuthConfig,
+  Discord: createDiscordOAuthConfig,
+  Dropbox: createDropboxOAuthConfig,
+  Facebook: createFacebookOAuthConfig,
+  GitHub: createGitHubOAuthConfig,
+  GitLab: createGitLabOAuthConfig,
+  Google: createGoogleOAuthConfig,
+  Notion: createNotionOAuthConfig,
+  Okta: createOktaOAuthConfig,
+  Patreon: createPatreonOAuthConfig,
+  Slack: createSlackOAuthConfig,
+  Spotify: createSpotifyOAuthConfig,
+  Twitter: createTwitterOAuthConfig,
+};
+
 /**
  * Allows for dynamic provider selection useful for testing.
- * In production, just use import and use the provider's OAuth 2.0 client creator.
+ * In production, just use import and use the provider's OAuth 2.0 config creator.
  *
  * @example
  * ```ts
- * import { createGitHubOAuth2Client } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
+ * import { createGitHubOAuthConfig } from "https://deno.land/x/deno_kv_oauth@$VERSION/src/providers/github.ts";
  *
- * const oauth2Client = createGitHubOAuth2Client();
+ * const oauthConfig = createGitHubOAuthConfig({
+ *   redirectUri: "/callback"
+ * });
  * ```
  */
-const provider = Deno.env.get("PROVIDER") ?? "GitHub";
-const createOAuth2ClientFn = {
-  Auth0: createAuth0OAuth2Client,
-  Discord: createDiscordOAuth2Client,
-  Dropbox: createDropboxOAuth2Client,
-  Facebook: createFacebookOAuth2Client,
-  GitHub: createGitHubOAuth2Client,
-  GitLab: createGitLabOAuth2Client,
-  Google: createGoogleOAuth2Client,
-  Notion: createNotionOAuth2Client,
-  Okta: createOktaOAuth2Client,
-  Patreon: createPatreonOAuth2Client,
-  Slack: createSlackOAuth2Client,
-  Spotify: createSpotifyOAuth2Client,
-  Twitter: createTwitterOAuth2Client,
-}[provider];
-
-if (createOAuth2ClientFn === undefined) {
-  throw new Error("Provider not found");
-}
+function getOAuthConfig() {
```

Comment:
I had to rejig this a little so that the test case could set the necessary env vars.